stories/utils/generate-widget-stories.js

```diff
@@ -158,6 +173,29 @@ export function generateReportBasedWidgetStories( {
 		},
 	};
 
+	// custom variants.
+	const customVariants = {};
+	Object.keys( additionalVariants ).forEach( ( name ) => {
+		const { data: variantData, options: variantOptions } = additionalVariants[ name ];
```

Comment:
If `variantOptions` is an array, we should have here the same type of validation as happens for `options` further above.